From 5c14b9998a7423806af5256d75c9241b0f0cd896 Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Wed, 1 Jul 2015 01:19:14 +0800
Subject: [PATCH] MLK-11204 ARM: imx: set int_mem_clk_lpm bit bit when
 entering wait mode on 6ul

This bit is used to keep the ARM Platform memory clocks enabled if
an interrupt is pending when entering low power mode. This bit should
always bet set when the CCM_CLPCR_LPM bits are set to 01(WAIT Mode) or
10 (STOP mode) without power gating.

Signed-off-by: Bai Ping <b51503@freescale.com>
---
 arch/arm/mach-imx/cpuidle-imx6ul.c | 3 +++
 arch/arm/mach-imx/pm-imx6.c        | 3 ++-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/cpuidle-imx6ul.c b/arch/arm/mach-imx/cpuidle-imx6ul.c
index d07375f..d9d445c 100644
--- a/arch/arm/mach-imx/cpuidle-imx6ul.c
+++ b/arch/arm/mach-imx/cpuidle-imx6ul.c
@@ -180,5 +180,8 @@ int __init imx6ul_cpuidle_init(void)
 	imx6ul_wfi_in_iram_fn = (void *)fncpy(wfi_iram_base + sizeof(*cpuidle_pm_info),
 		&imx6ul_low_power_idle, wfi_code_size);
 #endif
+
+	imx6q_set_int_mem_clk_lpm(true);
+
 	return cpuidle_register(&imx6ul_cpuidle_driver, NULL);
 }
diff --git a/arch/arm/mach-imx/pm-imx6.c b/arch/arm/mach-imx/pm-imx6.c
index aa18652..e1a45e2 100644
--- a/arch/arm/mach-imx/pm-imx6.c
+++ b/arch/arm/mach-imx/pm-imx6.c
@@ -479,7 +479,8 @@ void imx6q_set_int_mem_clk_lpm(bool enable)
 	if ((cpu_is_imx6q() && imx_get_soc_revision() >
 		IMX_CHIP_REVISION_1_1) ||
 		(cpu_is_imx6dl() && imx_get_soc_revision() >
-		IMX_CHIP_REVISION_1_0) || cpu_is_imx6sx()) {
+		IMX_CHIP_REVISION_1_0) || cpu_is_imx6sx() ||
+		cpu_is_imx6ul()) {
 		u32 val;
 
 		val = readl_relaxed(ccm_base + CGPR);
-- 
1.8.0

